mathics/core/expression.py

```diff
@@ -743,9 +742,30 @@ def flatten(
         else:
             return self
 
-    def evaluate(self, evaluation) -> typing.Union["Expression", "Symbol"]:
-        from mathics.core.evaluation import ReturnInterrupt
-
+    # When we allow 3.9 only, the return type could become type[Expression]
+    # See https://adamj.eu/tech/2021/05/16/python-type-hints-return-class-not-instance/
+    # Note that the return type is some subclass of BaseExpression, it could be
+    # a Real, an Expression, etc. It probably will *not* be a BaseExpression since
+    # the point of evaluation when there is not an error is to produce a concrete result.
+    def evaluate(
+        self, evaluation: typing.Type["BaseExpression"]
```

Comment:
``evaluation: typing.Type["Evaluation"]``

Reply:
Actually, I think it is exactly an "Evaluation", right?

Reply:
Yes